### PR TITLE
cli --min-packsize and  --file-read-concurrency

### PR DESCRIPTION
**Pull request by madsi1m**
What is the purpose of this change? What does it change?
--------------------------------------------------------

Would like to add the following in order to tune restic rather than hardcoding it.
--min-packsize
--file-read-concurrency

Was the change discussed in an issue or in the forum before?
------------------------------------------------------------

Nope

Checklist
---------

- [X] I have read the [Contribution Guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches)
- [X] I have enabled [maintainer edits for this PR](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
- [X] I have added tests for all changes in this PR
- [ ] I have added documentation for the changes (in the manual)
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (template [here](https://github.com/restic/restic/blob/master/changelog/TEMPLATE))
- [x] I have run `gofmt` on the code in all commits
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits)
- [x] I'm done, this Pull Request is ready for review

### CODE REVIEW

File: changelog/unreleased/pull-2655 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+Bugfix: Addition of --min-packsize and --file-read-concurrency
```
Comment (greatroar) [inline changelog/unreleased/pull-2655]: This is not a bugfix.

File: changelog/unreleased/pull-2655 [hunk 2]
```diff
@@ -0,0 +1,5 @@
...
+
+In order to tune restic for large installs we need to be able to configure --min-packsize and --file-read-concurrency. I the current release these numbers are hard coded.
```
Comment (greatroar) [inline changelog/unreleased/pull-2655]: This is written like a feature request. Also, "the current release" is the one that would incorporate this patch.

**Comment by greatroar** [comment]
Why have both options and environment variables? If only the latter were recognized, the changes could be made much more local. (I'm not a committer on this project, so take this as a suggestion only.)

**Comment by madsi1m** [comment]
Like other options in Restic, you can set environment variables. These environment variables can be overwritten by the cli variables. I assume this is so you can have many Restic endpoints/configurations per machine. It is a good idea that Restic have implemented and have copied it for these new variables .

File: changelog/unreleased/pull-2655 (hunk 1 above)
Comment (madsi1m) [reply changelog/unreleased/pull-2655]: modified

File: changelog/unreleased/pull-2655 (hunk 2 above)
Comment (madsi1m) [reply changelog/unreleased/pull-2655]: modified

**Comment by madsi1m** [comment]
Seems pullbot removed my changes. Created a new pull request https://github.com/restic/restic/pull/2671 and have disabled the bot

**Comment by MichaelEischer** [comment]
@madsi1m You should not develop features on the master branch. Please move your commits to another branch e.g. with `git checkout -b min-packsize`.

The pull-bot description states, that it resets your master branch to the upstream master (this repository), that is the bot just did what it should do.

**Comment by madsi1m** [comment]
Good tip!

**Pull request merged**